src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -0,0 +1,380 @@
+            {
+                var getters = new Delegate[1];
+                disposer = null;
+                using (var ch = _host.Start("CreateGetters"))
+                {
+                    if (activeOutput(0))
+                        getters[0] = MakeGetter(input);
+                    ch.Done();
+                    return getters;
+                }
+            }
+
+            public Func<int, bool> GetDependencies(Func<int, bool> activeOutput)
+            {
+                return col => activeOutput(0) && _inputColIndices.Any(i => i == col);
+            }
+
+            public RowMapperColumnInfo[] GetOutputColumns()
+            {
+                var info = new RowMapperColumnInfo[1];
+                info[0] = new RowMapperColumnInfo(_outputColName, _outputColType, null);
+                return info;
+            }
+
+            private static (ColumnType, TFDataType) GetOutputTypes(TFGraph graph, string columnName)
+            {
+                var tfoutput = new TFOutput(graph[columnName]);
+                var shape = graph.GetTensorShape(tfoutput);
+
+                int[] dims = new int[shape.NumDimensions];
+                for (int k = 0; k < shape.NumDimensions; k++)
+                    dims[k] = (int)(shape[k] == -1 ? 1 : shape[k]);
+
+                var type = TensorFlowUtils.Tf2MlNetType(tfoutput.OutputType);
+                return (new VectorType(type, dims), tfoutput.OutputType);
+            }
+
+            private static (string[], int[], bool[], TFShape[], TFDataType[]) GetInputMetaData(TFGraph graph, string[] source, ISchema inputSchema)
+            {
+                var tfShapes = new TFShape[source.Length];
+                var tfTypes = new TFDataType[source.Length];
+                var colNames = new string[source.Length];
+                var inputColIndices = new int[source.Length];
+                var isInputVector = new bool[source.Length];
+                for (int i = 0; i < source.Length; i++)
+                {
+                    colNames[i] = source[i];
+                    if (!inputSchema.TryGetColumnIndex(colNames[i], out inputColIndices[i]))
+                        throw Contracts.Except($"Column '{colNames[i]}' does not exist");
+
+                    var tfoutput = new TFOutput(graph[colNames[i]]);
+                    if (!TensorFlowUtils.IsTypeSupported(tfoutput.OutputType))
+                        throw Contracts.Except($"Input type '{tfoutput.OutputType}' of input column '{colNames[i]}' is not supported in TensorFlow");
+
+                    tfShapes[i] = graph.GetTensorShape(tfoutput);
+                    var type = inputSchema.GetColumnType(inputColIndices[i]);
+                    var shape = tfShapes[i].ToIntArray();
+                    int valCount = 1;
+                    for (int j = 1; j < shape.Length; j++)
+                        valCount *= shape[j];
```

Comment:
>shape[j] [](start = 36, length = 8)

Need to handle batch dimension properly because batch dimension is -1 most of the cases. The following check will fail in that case.